src/NzbDrone.Core/Qualities/QualityModelComparer.cs

```diff
@@ -16,17 +17,45 @@ public QualityModelComparer(Profile profile)
             _profile = profile;
         }
 
+        public int Compare(int left, int right, bool respectGroupOrder = false)
+        {
+            var leftIndex = _profile.GetIndex(left);
+            var rightIndex = _profile.GetIndex(right);
+
+            if (respectGroupOrder)
+            {
+                return leftIndex.CompareTo(rightIndex);
+            }
+
+            return Math.Floor(leftIndex).CompareTo(Math.Floor(rightIndex));
+        }
+
         public int Compare(Quality left, Quality right)
         {
-            int leftIndex = _profile.Items.FindIndex(v => v.Quality == left);
-            int rightIndex = _profile.Items.FindIndex(v => v.Quality == right);
+            return Compare(left, right, false);
+        }
+
+        public int Compare(Quality left, Quality right, bool respectGroupOrder)
+        {
+            var leftIndex = _profile.GetIndex(left);
+            var rightIndex = _profile.GetIndex(right);
+
+            if (respectGroupOrder)
+            {
+                return leftIndex.CompareTo(rightIndex);
+            }
 
-            return leftIndex.CompareTo(rightIndex);
+            return Math.Floor(leftIndex).CompareTo(Math.Floor(rightIndex));
         }
 
         public int Compare(QualityModel left, QualityModel right)
         {
-            int result = Compare(left.Quality, right.Quality);
+            return Compare(left, right, false);
+        }
+
+        public int Compare(QualityModel left, QualityModel right, bool respectGroupOrder)
+        {
+            int result = Compare(left.Quality, right.Quality, respectGroupOrder);
```

Comment:
> For Ordering we need to consider the qualities in one group to be equal.

For upgrade purposes we need to, but respecting the group ordering to make a decision on which to download makes sense. For example: preferring WEB-DL over Webrip would grab the WEB-DL if both are seen, but wouldn't upgrade Webrip to WEB-DL even if the cutoff is higher.